File: Java-Core/src/main/java/presto/javascript/JavaScriptNativeCall.java
package presto.javascript;

import presto.error.PrestoError;
import presto.error.SyntaxError;
import presto.runtime.Context;
import presto.statement.NativeCall;
import presto.type.IType;
import presto.type.VoidType;
import presto.utils.CodeWriter;
import presto.value.IValue;

public class JavaScriptNativeCall extends NativeCall {

	JavaScriptStatement statement;
	
	public JavaScriptNativeCall(JavaScriptStatement statement) {
		this.statement = statement;
	}
	
	@Override
	public void toDialect(CodeWriter writer) {
		writer.append("JavaScript: ");
		statement.toDialect(writer);
	}
	
	@Override
	public IType check(Context context) throws SyntaxError {
		return VoidType.instance(); // TODO
	}
	
	@Override
	public IValue interpret(Context context) throws PrestoError {
		throw new RuntimeException("Should never get there!");
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/javascript/JavaScriptNativeCall.js
var NativeCall = require("../statement/NativeCall").NativeCall;

function JavaScriptNativeCall(statement, module) {
	NativeCall.call(this);
	this.statement = statement;
	return this;
}

JavaScriptNativeCall.prototype = Object.create(NativeCall.prototype);
JavaScriptNativeCall.prototype.constructor = JavaScriptNativeCall;

JavaScriptNativeCall.prototype.toString = function() {
	return this.statement.toString();
};

JavaScriptNativeCall.prototype.check = function(context) {
	return this.statement.check(context);
};

JavaScriptNativeCall.prototype.interpret = function(context, returnType) {
	return this.statement.interpret(context, returnType);
};

JavaScriptNativeCall.prototype.toDialect = function(writer) {
    writer.append("JavaScript: ");
    this.statement.toDialect(writer);
};

exports.JavaScriptNativeCall = JavaScriptNativeCall;

